### PR TITLE
Rework ZHA group adds and removes

### PR DESCRIPTION
**Pull request by dmulcahey**
## Proposed change

This PR fixes the grouping UI so that it works based on endpoints and not based on devices. 

**REQUIRES: https://github.com/home-assistant/core/pull/34583**

A full explanation can be found in the description of the linked PR.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/zha/zha-device-endpoint-data-table.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+              filterable: false,
+              width: "50%",
+              template: (entities) => html`
+                <div>
+                  ${entities.length
+                    ? entities.map(
+                        (entity) =>
+                          html`<div
+                            class="mdc-data-table__cell table-cell-text"
+                          >
```
Comment (bramkragten) [inline src/panels/config/zha/zha-device-endpoint-data-table.ts:125]: Why create multiple cells in a cell?

File: src/panels/config/zha/zha-group-page.ts [hunk 2]
```diff
@@ -20,46 +20,51 @@ import {
...
+  ZHADeviceEndpoint,
 } from "../../../data/zha";
 import "../../../layouts/hass-error-screen";
 import "../../../layouts/hass-subpage";
 import { HomeAssistant } from "../../../types";
 import "../ha-config-section";
 import { formatAsPaddedHex } from "./functions";
 import "./zha-device-card";
-import "./zha-devices-data-table";
+// eslint-disable-next-line import/no-duplicates
```
Comment (bramkragten) [inline src/panels/config/zha/zha-group-page.ts:32]:
```suggestion
```

File: src/panels/config/zha/zha-group-page.ts [hunk 3]
```diff
@@ -20,46 +20,51 @@ import {
...
+import "./zha-device-endpoint-data-table";
+// eslint-disable-next-line import/no-duplicates
```
Comment (bramkragten) [inline src/panels/config/zha/zha-group-page.ts:34]:
```suggestion
```

File: src/panels/config/zha/zha-device-endpoint-data-table.ts (hunk 1 above)
Comment (dmulcahey) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts:125]: image

It's how I got the associated entities to render correctly like this, including the elided text. Does that make sense?

File: src/panels/config/zha/zha-group-page.ts (hunk 2 above)
Comment (dmulcahey) [reply src/panels/config/zha/zha-group-page.ts]: doing this broke CI... what am I missing?

Comment (dmulcahey) [reply src/panels/config/zha/zha-group-page.ts]: nvm figured it out

**Comment by dmulcahey** [comment]
@bramkragten any chance we can get this done before the beta? If not, I'll have to revert the PR on core.

**Comment by bramkragten** [comment]
Yeah should not be a problem

File: src/panels/config/zha/zha-device-endpoint-data-table.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts:125]: We should not do that, also, what happens when there are 6 entities?

File: src/panels/config/zha/zha-device-endpoint-data-table.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                            ${entity.name || entity.original_name}
+                          </div>`
+                      )
+                    : html`
+                        <p>
+                          This endpoint has no associated entities
+                        </p>
+                      `}
```
Comment (bramkragten) [inline src/panels/config/zha/zha-device-endpoint-data-table.ts:133]:
```suggestion
                    : This endpoint has no associated entities}
```

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/zha/zha-device-endpoint-data-table.ts (hunk 1 above)
Comment (dmulcahey) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts:125]: We don't have anything where that would be possible yet... if it comes to pass and we have a situation like that I can elide the list too. Do you have any suggestions?

Comment (bramkragten) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts:125]: Add `style="overflow: hidden; text-overflow: ellipsis;"` instead of the classes.

Comment (bramkragten) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts]: So it is not possible that there are more than 3 entities associated with a groupable device?

Comment (bramkragten) [reply src/panels/config/zha/zha-device-endpoint-data-table.ts]: Updated it to make sure: 
image

**Pull request merged**